### ISSUE TITLE
User info for message to self displays wrong user profile

### ISSUE THREAD

**Issue by jatoko** [RocketChat/Rocket.Chat#20538]
### Description:

In a direct chat with oneself, if one clicks on "User info", a wrong, but always the same user is displayed for all users.

### Steps to reproduce:

1. Start direct chat with own account.
2. Click on User Info.
3. Wrong user displayed.

### Expected behavior:

See my own account profile.

### Actual behavior:

See another users' profile; it is the same "wrong" profile for all users.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.10.5
- Operating System: Linux Ubuntu
- Deployment Method: docker
- Number of Running Instances: 1 
- DB Replicaset Oplog: 
- NodeJS Version: v12.18.4
- MongoDB Version: 4.0.22

### Client Setup Information

- Desktop App or Browser Version: 1.21.0-alpha.4235
- Operating System: Windows, MacOS

**Comment by yash-rajpal** [comment]
I am not able to reproduce this, can you try on open workspace of rocket chat, [here](https://open.rocket.chat/home) ?
If you can provide more info or provide a video, that would be helpful. :)

**Comment by sketch242** [comment]
I saw this on the mac desktop client 3.1.1 with server version 3.11.1.  A few different users on our private server reported this, and so far everyone sees the same user's info.

I logged in with a browser to see if it was limited to the desktop client and when I initially searched for my name, it created a new DM chat with myself with no history, where it no longer happens.  But it still happens in the old DM chat history.
